content/articles/build-autocorrect-feauture-using-nlp-with-python/index.md

```diff
@@ -0,0 +1,285 @@
+date: 2021-11-13T00:00:00-2:30
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/build-autocorrect-feature-using-nlp-with-python/hero.jpg
+    alt: Natural Language processing example image
+---
+One application of Natural Language Processing (NLP) is the Autocorrect function. This feature works on every smartphone keyboard regardless of the brand, it is specially programmed to generalize all the correct words that are in the vocabulary and tries to find the most similar words to those words, not in the vocabulary.
+<!--more-->
+To understand how it works, will cover Natural Language Processing in this article and how we can use it with Python to build the autocorrect feature showing all the steps involved. 
+
+### Table of contents
+- [Table of contents](#table-of-contents)
+- [Prerequisites](#prerequisites)
+- [Natural Language Processing(NLP)](#natural-language-processingnlp)
+- [Autocorrect Feature](#autocorrect-feature)
+- [Build the Autocorrect Feature](#build-the-autocorrect-feature)
+  - [Installing Libraries](#installing-libraries)
+  - [Reading text file(dictionary)](#reading-text-filedictionary)
+  - [Frequency and Probability of Dictionary Words](#frequency-and-probability-of-dictionary-words)
+  - [Implement 4 edit word functions](#implement-4-edit-word-functions)
+  - [Finding Similar/Corrected Word](#finding-similarcorrected-word)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+To follow along with this tutorial, the reader should:
+- Have a prior understanding of the basics of Natural Language Processing and Machine Learning concepts.
+- Have a basic understanding of Python and the various python libraries used in Natural language processing.
+- Know how to use Pycharm or any other IDE or code editor for running Python.
+
+To download and install the Pycharm IDE on your PC, go to this [site](https://www.jetbrains.com/pycharm/download/#section=windows).
+
+
+### Natural Language Processing(NLP)
+Natural Language Processing (NLP) is a branch of computer science and artificial intelligence that focuses on allowing computers to understand and process human natural language.. NLP is a programming language that allows computers to evaluate and interpret large volumes of natural language data.
+It paves the door for more interactivity and productivity in a variety of fields:
+- Search Autocorrect and Autocomplete.
+- Language Translation and Grammar checkers.
+- Chatbots and Social Media monitoring.
+- Email filtering and Voice Assistants.
+  
+We'll look at how it's employed in autocorrection systems in this tutorial.
+
+### Autocorrect Feature
+The Autocorrect model is completely based on Natural Language Processing (NLP), and it is programmed to correct spellings and errors while inputting text and tries to locate the most comparable related words, as the name suggests. It works by comparing the words in the vocabulary dictionary and the typed words on the keyboard. 
```

Comment:
Change to " The Autocorrect model is completely based on Natural Language Processing (NLP). It is programmed to correct spellings and errors while inputting text and locating the most comparable related words, as the name suggests. It compares the words in the vocabulary dictionary and the typed words on the keyboard. "